### ISSUE TITLE
Improve SNOMED match rate from 0.889 to 0.95+ (vector index + LCA)

### ISSUE THREAD

**Issue by ricocopapa** [ricocopapa/vet-snomed-rag#2]
## Context

v2.1 delivered SNOMED Match **0.584 → 0.889 (+52.2%, 5/9 → 8/9)** via a 4-stage improvement (MRCM pattern relaxation / BGE reranker / MRCM direct mapping / metrics fix + GI_VOMIT direct mapping). See the v2.1 Release Notes for details: https://github.com/ricocopapa/vet-snomed-rag/releases/tag/v2.1

The remaining gap to 1.000 is **not** algorithmic — it is a single gold-label mismatch (S03 `OR_LAMENESS_FL_L`, documented in `GOLD_AUDIT.md §7`). Isolated overfitting to that one case is explicitly rejected.

Beyond that, v2.1 still operates on a **9-gold synthetic benchmark**. Pushing SNOMED Match materially higher — and credible on a realistic dataset — requires a different class of improvement: vector-index quality and LCA-weighted ranking.

**Related**: #1 (real recordings), #5 (multi-reviewer gold κ)

## Problem

Strategy B of v2.1 produced the crucial finding: **BGE Reranker alone cannot rescue candidates missing from ChromaDB Top-20**. When the retriever's semantic embedding for `"cornea clarity"` pulls `Opacity of cornea (1099051000119104)` closer than `Corneal edema (27194006)`, no reranker weight fixes the gap at ranking time.

v2.1 worked around this with MRCM direct mapping (bypass RAG), which is correct for unambiguous single-concept fields but does not scale to ambiguous or multi-concept fields. To reach 0.95+ on a larger benchmark, the retriever itself must improve.

## Reproduction

```bash
# v2.1 baseline
python scripts/evaluate_e2e.py \
  --input-mode text \
  --input-dir data/synthetic_scenarios/ \
  --snomed-mode synonym
# SNOMED Match 0.889 (8/9); remaining FAIL is S03 OR_LAMENESS_FL_L gold mismatch

# Reranker limit demonstration (v2.1 strategy B finding)
python scripts/snomed_strategy_b_benchmark.py
# Observe: BGE smoke-test reverses Corneal edema > Post-surgical haze, but E2E tagging
# still UNMAPPED because the correct concept is not in Top-20 from ChromaDB.
```

## Proposed Approach

**Option A (preferred) — ChromaDB index retraining with domain-adapted embeddings**
- Switch from `all-MiniLM-L6-v2` to a veterinary-domain-adapted model (e.g. fine-tune on SNOMED-Vet FSN pairs)
- Rebuild the 414K-concept index; measure Top-20 recall for known hard cases
- Effort: 6–10h (index rebuild dominates)

**Option B — LCA-weighted re-ranking (strategy C-2 from v2.1 design doc)**
- Use SNOMED `is-a` relationships to penalize candidates far from the expected semantic tag
- Build IS-A graph from `data/snomed_ct_vet.db`, compute LCA distance in post-processing
- Effort: 6–10h; requires DB schema work

**Option C — BM25 tuning + query expansion**
- Introduce true BM25 on the SQL side (replace LIKE); expand queries with field-code-derived synonyms
- Tune RRF α/β weights empirically
- Effort: 3–5h; moderate gains expected

## Acceptance Criteria

- [ ] SNOMED Match Text ≥ 0.950 on the expanded (post-#1, post-#5) gold set — not on the current 9-sample toy set
- [ ] Top-20 recall on SNOMED VET 414K concepts improves for hard cases (measure OPH_CORNEA class)
- [ ] No regression in Precision (≥ 0.891) or Recall (≥ 0.772) at text mode
- [ ] Latency p95 text mode stays ≤ 40s (allowing +5s for richer ranking)
- [ ] `pytest tests/` preserved at 85 passed / 1 skipped

## Estimated Effort

**8–16 hours** — index retraining (A) or LCA (B) dominates. Can be split into sub-issues once approach is chosen.

## Milestone & Labels

- Milestone: `v2.2`
- Labels: `v2.2`, `rag`, `enhancement`
